configure.ac

```diff
@@ -475,7 +475,9 @@ if test "x$CXXFLAGS_overridden" = "xno"; then
   AX_CHECK_COMPILE_FLAG([-Wself-assign],[NOWARN_CXXFLAGS="$NOWARN_CXXFLAGS -Wno-self-assign"],,[[$CXXFLAG_WERROR]])
   AX_CHECK_COMPILE_FLAG([-Wunused-local-typedef],[NOWARN_CXXFLAGS="$NOWARN_CXXFLAGS -Wno-unused-local-typedef"],,[[$CXXFLAG_WERROR]])
   AX_CHECK_COMPILE_FLAG([-Wimplicit-fallthrough],[NOWARN_CXXFLAGS="$NOWARN_CXXFLAGS -Wno-implicit-fallthrough"],,[[$CXXFLAG_WERROR]])
-  AX_CHECK_COMPILE_FLAG([-Wdeprecated-copy],[NOWARN_CXXFLAGS="$NOWARN_CXXFLAGS -Wno-deprecated-copy"],,[[$CXXFLAG_WERROR]])
+  if test x$suppress_external_warnings == xno ; then
+    AX_CHECK_COMPILE_FLAG([-Wdeprecated-copy],[NOWARN_CXXFLAGS="$NOWARN_CXXFLAGS -Wno-deprecated-copy"],,[[$CXXFLAG_WERROR]])
```

Comment:
I'm almost certain @hebasto already made this comment, but I must have imagined it. Can you move this up to below `AX_CHECK_COMPILE_FLAG([-Wdocumentation]`... . Then this is ok.

Reply:
> inside the existing if test x$suppress_external_warnings != xno ; then check.

That wouldn't work. The check needs to be `== xno`, not `!= xno`. Alternatively it can be `!= xyes`, I guess.

Also, I want to keep the `NOWARN_CXXFLAGS` in one section and not mix it with the `WARN_CXXFLAGS` section

Reply:
Right, in this case we're removing a suppression only when enabling other suppressions. Changing this to `!=yes` would better match how we generally do this in configure

Reply:
Thanks, pushed a fix. Didn't re-test.

```diff
-  if test x$suppress_external_warnings == xno ; then
+  if test x$suppress_external_warnings != xyes ; then